src/aws.ts

```diff
@@ -0,0 +1,314 @@
+import { Table } from "./table";
+import {
+  UpdateItemInput,
+  UpdateItemOutput,
+} from "typesafe-dynamodb/lib/update-item";
+import { PutItemInput, PutItemOutput } from "typesafe-dynamodb/lib/put-item";
+import { ScanInput, ScanOutput } from "typesafe-dynamodb/lib/scan";
+import { QueryInput, QueryOutput } from "typesafe-dynamodb/lib/query";
+import { GetItemInput, GetItemOutput } from "typesafe-dynamodb/lib/get-item";
+import {
+  DeleteItemInput,
+  DeleteItemOutput,
+} from "typesafe-dynamodb/lib/delete-item";
+import { TableKey } from "typesafe-dynamodb/lib/key";
+import { JsonFormat } from "typesafe-dynamodb";
+import { CallExpr, isObjectLiteralExpr, ObjectLiteralExpr } from "./expression";
+import { ASL, isASL, Task } from "./asl";
+import { CallContext } from "./context";
+import { VTL } from "./vtl";
+
+import type { DynamoDB as AWSDynamoDB } from "aws-sdk";
+
+type Item<T extends Table<any, any, any>> = T extends Table<infer I, any, any>
+  ? I
+  : never;
+
+type PartitionKey<T extends Table<any, any, any>> = T extends Table<
+  any,
+  infer PK,
+  any
+>
+  ? PK
+  : never;
+
+type RangeKey<T extends Table<any, any, any>> = T extends Table<
+  any,
+  any,
+  infer SK
+>
+  ? SK
+  : never;
+
+/**
+ * The `AWS` namespace exports functions that map to AWS Step Functions AWS-SDK Integrations.
```

Comment:
My thought with `$AWS` was that it would mirror the aws-sdk:

```ts
import * as AWS from. "aws-sdk";
```

I'm hoping that it could serve as the "functionless" aws-sdk for any integration that uses the exact API, as opposed to optimized integrations  that diverge from the AWS SDK, e.g. Appsync DynamoDB resolvers.

Reply:
I feel like we'd quickly have a web of methods that do and don't work in specific cases.  For example, lambda has an option to return the whole https payload in step functions, block and wait on a token, or return just the lambda response payload (return type), these are mostly all step function specific things.

Keeping integration local utilities would reduce the impact of divergence (pure typescript will not diverge, just the service specific utilities). 

We could abstract out shared utilities once we know the patterns don't change.

Being global may just be early optimization?

Reply:
We can resolve this by having the $AWS always be exactly the raw sdk. Use `$AWS.Lambda.Invoke(func)` instead of `func()` to get the whole payload. A

Reply:
https://github.com/sam-goodwin/functionless/issues/76